### PR TITLE
Workaround to Oracle11/ojdbc6 missing getSchema API.

### PR DESCRIPTION
**Pull request by glenna-m**
TeamCity testing identified an issue with Oracle11 and the new tableExists API.  TableExists(tableName) uses connection.getSchema() to determine the current schema, but that API was not added until  JDK 1.7.    Oracle 11 uses the ojdbc6 driver which does not support it. 

I tried quite a lot of things to address the problem.  This checkin takes the best approach I was able to find.  I use the reflections API to see if getSchema is defined on the connection's class.  If it is defined, it is invoked.  If it is not defined and this is Oracle, it will default to using the username as the schema (as Oracle does), warning that is what we are doing and recommending either upgrading to a newer driver or use the tableExists(tableName, schemaName) API and specify your schema.   If an exception is thrown, a DatabaseException warn that we were unable to determine the schema and also recommends upgrading or using the tableExists(tableName, schemaName) API.

### CODE REVIEW

File: src/main/java/com/github/susom/database/DatabaseImpl.java [hunk 1]
```diff
@@ -206,14 +207,29 @@ public void dropTableQuietly(/*@Untainted*/ @Nonnull String tableName) {
...
-    } catch (SQLException exc) {
-      throw new DatabaseException("Unable to look up table " + tableName +
-        " with default db connection schema & catalog: " + exc.getMessage(),
-        exc);
+      // Use reflections to see if connection.getSchema API exists. It should exist for any JDBC7 or later implementation
+      // We still support Oracle 11 with odbc6, however, so we can't assume it's there.
+      Method method = connection.getClass().getMethod("getSchema");
+      if (method != null) {
+        schemaName = (String) method.invoke(connection);
+      } else if ("oracle".equals(this.flavor().toString())) {
```
Comment (garricko) [inline src/main/java/com/github/susom/database/DatabaseImpl.java]: Cleaner to say `flavor() == Flavor.oracle` since these are defined as enumerations

**Comment by jpallas** [comment]
Not to be too radical, but … this library requires JDK8.  Why can't we require ojdb8?

**Comment by smalunjk** [comment]
will this ever return null?

schemaName = (String) method.invoke(connection);

File: src/main/java/com/github/susom/database/DatabaseImpl.java (hunk 1 above)
Comment (glenna-m) [reply src/main/java/com/github/susom/database/DatabaseImpl.java]: This change will be in next check in

**Comment by glenna-m** [comment]
> Not to be too radical, but … this library requires JDK8. Why can't we require ojdbc8?

The issue is that Maven's Oracle 11 requires ojdbc6.  We'd need to require Oracle 12 to do that.

**Comment by glenna-m** [comment]
> will this ever return null?
> 
> schemaName = (String) method.invoke(connection);

It usually throws an exception, but I guess that's possible.  I will check for it.

**Comment by jpallas** [comment]
> > Not to be too radical, but … this library requires JDK8. Why can't we require ojdbc8?
> 
> The issue is that Maven's Oracle 11 requires ojdbc6. We'd need to require Oracle 12 to do that.

Maybe we can fix that.  @garricko, what distinction is intended here?  The Oracle version matrix at https://www.oracle.com/database/technologies/faq-jdbc.html says that all the versions of the JDBC driver support all the versions of the database, although it recommends that the driver version be at least as high as the database version.

File: src/main/java/com/github/susom/database/DatabaseImpl.java [hunk 2]
```diff
@@ -27,9 +28,6 @@
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-/*>>>
-import org.checkerframework.checker.tainting.qual.Untainted;
-*/
```
Comment (garricko) [inline src/main/java/com/github/susom/database/DatabaseImpl.java]: I don't think you want to remove this, as it is used by the static analysis tool (Checker Framework).

**Comment by garricko** [comment]
> Maybe we can fix that. @garricko, what distinction is intended here? The Oracle version matrix at https://www.oracle.com/database/technologies/faq-jdbc.html says that all the versions of the JDBC driver support all the versions of the database, although it recommends that the driver version be at least as high as the database version.

Oracle 11 doesn't distribute ojdbc8.jar with the driver bundle as far as I can tell. I know the documentation states you can mix and match driver and database versions, but Oracle support told us not to do that when we found and reported compatibility problems.

**Pull request merged**